### PR TITLE
update to CockroachDB v22.2.19

### LINKED ISSUE #2017: update to Cockroach 22.2.0
Cockroach have just released v22.2.0 with Go 1.19.1.  It'd be good to update to this as soon as convenient.

- [ ] update garbage-compactor to build v22.2.0
    - [ ] need Go 1.19.1 IPS package
    - [ ] need newer Node IPS package (>=16.10)
- [ ] update omicron to use the v22.2.0 build

### PR DESCRIPTION
**Pull request by iliana**
Closes #2017.

This is a draft to prevent merging until R9 goes out, but feel free to review.

### CODE REVIEW

File: tools/ci_download_cockroachdb [hunk 1]
```diff
@@ -101,12 +98,17 @@ function fail
...
+				CIDL_SHA256="$CIDL_SHA256_DARWIN_ARM64"
+			else
+				CIDL_BUILD="darwin-10.9-amd64"
+				CIDL_SHA256="$CIDL_SHA256_DARWIN_AMD64"
+			fi
 			CIDL_SUFFIX="tgz"
-			CIDL_SHA256="$CIDL_SHA256_DARWIN"
 			CIDL_URL_BASE="$CIDL_URL_COCKROACH"
 			CIDL_ASSEMBLE="do_assemble_official"
 			;;
```
Comment (iliana) [inline tools/ci_download_cockroachdb:114]: also, a small gift for @david-crespo

**Comment by citrus-it** [comment]
> uploaded somewhere more official than my personal web server. For now, a draft branch to kick off the full gauntlet of CI testing.

Perhaps put these in an S3 bucket like the clickhouse artefacts? There is already a bucket that could be renamed and used (and possibly some of the older things cleaned out)

**Comment by iliana** [comment]
Yeah I can poke around and see if I can find that S3 bucket or one more appropriate in one of the company accounts.

File: tools/ci_download_cockroachdb [hunk 2]
```diff
@@ -101,12 +96,17 @@ function fail
 function configure_os
 {
 	echo "current directory: $PWD"
-	echo "configuring based on OS: \"$1\""
+	echo "configuring based on OS: \"$1\" \"$2\""
 	case "$1" in
 		darwin*)
-			CIDL_BUILD="darwin-10.9-amd64"
+			if [[ "$2" = "aarch64" ]]; then
```
Comment (citrus-it) [inline tools/ci_download_cockroachdb:102]: Technically should be `==`, although `=` works.
On my mac, `$HOSTTYPE` is arm64, not aarch64.
```
pptang:omicron:main% bash

bash-3.2$ echo $OSTYPE
darwin23
bash-3.2$ echo $HOSTTYPE
arm64
bash-3.2$
```

Comment (david-crespo) [reply tools/ci_download_cockroachdb:102]: hmmm

```
$ bash
bash-5.2$ echo $HOSTTYPE
aarch64
bash-5.2$ echo $OSTYPE
darwin23.2.0
```

Comment (david-crespo) [reply tools/ci_download_cockroachdb:102]: That should work. Looks like we could also use `uname -m`, which returns `arm64` consistently, but that is more complicated than sticking to `$HOSTTYPE`.

Comment (iliana) [reply tools/ci_download_cockroachdb:102]: Apple's refusal to ship a GPLv3-licensed Bash strikes again!!

File: nexus/db-queries/src/db/datastore/cockroachdb_settings.rs [hunk 3]
```diff
@@ -153,10 +153,20 @@ mod test {
...
 
         let settings = datastore.cockroachdb_settings(&opctx).await.unwrap();
-        // With a fresh cluster, this is the expected state
-        let version = CockroachDbClusterVersion::NEWLY_INITIALIZED.to_string();
-        assert_eq!(settings.version, version);
-        assert_eq!(settings.preserve_downgrade, "");
+        // `version` will either match `NEWLY_INITIALIZED` while
+        // running tests normally, or `POLICY` while running tests with
+        // `CRDB_SEED_USE_PREV=yes`.
+        let version: CockroachDbClusterVersion =
```
Comment (iliana) [inline nexus/db-queries/src/db/datastore/cockroachdb_settings.rs:159]: I believe the changes to this test would land okay in R9 ahead of this PR.

File: nexus/src/app/rack.rs [hunk 4]
```diff
@@ -241,22 +241,41 @@ impl super::Nexus {
...
-            )
-            .await
-            .internal_context(
-                "setting `cluster.preserve_downgrade_option` \
-                for rack initialization",
-            )?;
+        blueprint.cockroachdb_setting_preserve_downgrade =
+            if cockroachdb_settings.preserve_downgrade.is_empty() {
+                // Set the option to the current policy in both the database and
+                // the blueprint.
```
Comment (iliana) [inline nexus/src/app/rack.rs:247]: Same here, I think this is fine to land in R9 as well. The else branch here should never get hit in normal rack initialization, we only expect this setting to be non-empty when crdb-seed is running in `CRDB_SEED_USE_PREV` mode.

**Comment by iliana** [comment]
Closing in favor of #6396 and its follow-up PR.

**Pull request closed without merging**